hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -870,6 +873,18 @@ TypedProperties buildFileGroupReaderProperties(HoodieMetadataConfig metadataConf
     props.setProperty(
         DISK_MAP_BITCASK_COMPRESSION_ENABLED.key(),
         Boolean.toString(commonConfig.isBitCaskDiskMapCompressionEnabled()));
+
+    // Enable HFile block caching for all MDT partitions regardless of the configuration to avoid rescanning the HFiles for better performance
+    props.setProperty(HoodieReaderConfig.HFILE_BLOCK_CACHE_ENABLED.key(), "true");
```

Comment:
Let's only enable HFile caching for MDT files partition when used by MDT-based file system view for parity as before.

Reply:
Fixed.